### PR TITLE
S3: Updated retry logic for S3Stream.signAndRequest

### PR DESCRIPTION
**Pull request by bpiper**
References #2535

### CODE REVIEW

**Review by seglo — COMMENTED** [review]
Looking good. Just some notes on `Settings` boilerplate.

File: s3/src/main/scala/akka/stream/alpakka/s3/settings.scala [hunk 1]
```diff
@@ -238,6 +238,7 @@ final class S3Settings private (
     val listBucketApiVersion: ApiVersion,
     val forwardProxy: Option[ForwardProxy],
     val validateObjectKey: Boolean,
+    val retrySettings: RetrySettings,
```
[the remark this answers is not in the document]
Comment (bpiper) [reply s3/src/main/scala/akka/stream/alpakka/s3/settings.scala:355]: Oops, looks like I missed that stuff in a previous PR as well.  Should hopefully all be there now.

File: s3/src/main/scala/akka/stream/alpakka/s3/settings.scala [hunk 2]
```diff
@@ -229,6 +230,119 @@ object AccessStyle {
...
+  def getMaxBackoff: JavaDuration = JavaDuration.ofNanos(maxBackoff.toNanos)
+
+  /** Java API */
+  def getRandomFactor: Double = randomFactor
+
+  def withMaxRetries(value: Int): RetrySettings = copy(maxRetries = value)
+
+  def withMinBackoff(value: FiniteDuration): RetrySettings = copy(minBackoff = value)
+
+  def withMinBackoff(value: JavaDuration): RetrySettings =
```
Comment (seglo) [inline s3/src/main/scala/akka/stream/alpakka/s3/settings.scala:254]: Can you decorate these java setters with `/** Java API */` too?

**Review by seglo — COMMENTED** [review]
Some minor docs updates.

File: s3/src/main/scala/akka/stream/alpakka/s3/settings.scala (hunk 2 above)
Comment (bpiper) [reply s3/src/main/scala/akka/stream/alpakka/s3/settings.scala:255]: Sure.  I just added it to the ones that take `java.time.Duration` but let me know if I should add it to the others too.

Comment (seglo) [reply s3/src/main/scala/akka/stream/alpakka/s3/settings.scala:255]: I think that's all of them!

**Comment by seglo** [comment]
Looks like there's a pile of MiMa exceptions that need to be added for the new APIs. The build should tell you the exact rules you need to add. LMK if you need help creating these.

https://travis-ci.com/github/akka/alpakka/jobs/449213845#L1126

**Comment by bpiper** [comment]
@seglo any advice on resolving the binary compatibility issues? They're kind of a non issue since nobody's likely to be using those case class functions on the settings classes, but not sure if it's better to suppress MiMa somehow or preserve binary compatibility in some not-too-ugly way.

**Review by seglo — APPROVED** [review]
LGTM

**Comment by seglo** [comment]
Thanks!

**Pull request merged**